server/src/providers/blockchain.ts

```diff
@@ -301,12 +307,8 @@ export class blockchain {
    * @return {Promise<void>}
    */
   async checkAddressesClaimed (arrayOfAddresses: string[]): Promise<void> {
-    // there could be duplicates, so need to get unique values
-    // new Set([...]) -> will return unique values from received array
-    const uniqueAddresses = [...new Set(arrayOfAddresses)]
-
     // check multiple addresses exists and create new records in case if not exist by one db query
-    const { nonExistedCount } = await AddressesClaimedProvider.checkIfExistsMultiple(uniqueAddresses)
+    const { nonExistedCount } = await AddressesClaimedProvider.checkIfExistsMultiple(arrayOfAddresses)
```

Comment:
1. why you don't use uniqueAddresses  here ?
2. as we're using lodash, we could use it's uniq() function

Reply:
3. wallet addresses are case-insensitive, so 0xABCDxxxx and 0xaBCcdxxxx are the same addresses. So i think the most better is to use uniqBy(arrayOfAddresses , toLower) (form lodash)

Reply:
1. I move uniq to the updateClaimEvents (line 380)
2. done
3. done

Reply:
1. i saw that but why non-unique addresses are now passed to AddressesClaimedProvider.checkIfExistsMultiple() ? why you decided do not pass the unique ones like it was before ?